docs/using_pyodide_from_webworker.md

```diff
@@ -0,0 +1,110 @@
+
+For example:
+
+```
+var pyodideWorker = new Worker('./webworker.js')
+
+pyodideWorker.onerror = (e) => {
+  console.log(`Error in pyodideWorker at ${e.filename}, Line: ${e.lineno}, ${e.message}`)
+}
+
+pyodideWorker.onmessage = (e) => {
+  const {results, error} = e.data
+  if (results) {
+    console.log('pyodideWorker return results: ', results)
+  } else if (error) {
+    console.log('pyodideWorker error: ', error)
+  }
+}
+
+const data = {
+  A_rank: [0.8, 0.4, 1.2, 3.7, 2.6, 5.8],
+  python:
+    'import statistics\n' +
+    'from js import A_rank\n' +
+    'statistics.stdev(A_rank)'
+}
+
+pyodideWorker.postMessage(data)
+
+```
+
+## Loading packages
+
+Packages referenced from your python script will be automatically downloaded
+the first time they are encountered. Please note that some of the larger
+packages such as `numpy` or `pandas` may take several seconds to load.
+Subsequent uses of these packages will not incur the download overhead of the
+first run, but there is still some time required for the `import` in python
+itself.
+
+If you would like to pre-load some packages, or the automatic package loading
+does not work for you for some reason, you may modify the `webworker.js` source
+to load some specific packages as described in
+[Using Pyodide directly from Javascript](using_pyodide_from_javascript.md).
+
+For example, to always load packages `numpy` and `pytz`, you would insert the
+line `self.pyodide.loadPackage(['numpy', 'pytz']).then(() => {` as shown below:
+```
+self.languagePluginUrl = 'http://localhost:8000/'
+importScripts('./pyodide.js')
+
+var onmessage = function(e) { // eslint-disable-line no-unused-vars
+  languagePluginLoader.then(() => {
+    self.pyodide.loadPackage(['numpy', 'pytz']).then(() => {
+      const data = e.data;
+      const keys = Object.keys(data);
+      for (let key of keys) {
+        if (key !== 'python') {
+          // Keys other than python must be arguments for the python script.
+          // Set them on self, so that `from js import key` works.
+          self[key] = data[key];
+        }
+      }
+      self.pyodide.runPythonAsync(data.python, () => {})
+          .then((results) => { self.postMessage({results}); })
+          .catch((err) => {
+            // if you prefer messages with the error
+            self.postMessage({error : err.message});
+            // if you prefer onerror events
+            // setTimeout(() => { throw err; });
+          });
+    });
+  });
+}
+```
+
+## Caveats
+
+Using a web worker is advantageous because the python code is run a separate
+thread from your main UI, and hence does not impact your application's
+responsiveness. There are some limitations, however. Because web workers are
+each in their own virtual machine, there is no way to share the python
+interpreter or the downloaded packages between multiple web workers or with
+your main thread. Similarly, because of the separate VMs, you cannot share
```

Comment:
There are ways -- WebAssembly modules can be shared with webworkers -- we just need a convenient way to make that work within Pyodide.

Maybe instead say: `At present, Pyodide does not support sharing the Python interpreter and packages between multiple web workers or with your main thread`.